### DISCUSSION TITLE
OTel SDK Variables are not being applied

### DISCUSSION THREAD

**Discussion by arielvalentin** [istio/istio#49298] [Q&A] [answered]
Hello there from the GitHub Observability Team! 

We are using the OpenTelemetry tracing provider to generate traces, however it seems that the OTEL_SERVICE_NAME and OTEL_RESOURCE_ATTRIBUTES are not being picked up or passed along to the envoy tracing extension.

Is there an explicit configuration we must set in order to have envoy use the default SDK environment variables?

**Comment by howardjohn** [comment, marked as answer]
Coincidentally, this was just merged _today_: https://github.com/istio/istio/pull/49204. It does require explicit opt in (though I am wondering, @joaopgrassi, if it might make sense to enable the environment one by default actually... given the user explicitly opted in via OTEL_RESOURCE_ATTRIBUTES really).

@arielvalentin note the env var needs to be in the sidecar though, not the application.

**Reply by arielvalentin** [reply]
Thank you so much for the quick response! We will keep an eye out for the next release.
